libcaf_net/caf/net/http/client.cpp

```diff
@@ -162,14 +162,14 @@ class client_impl : public client {
     ptrdiff_t consumed = 0;
     if (mode_ == mode::read_header) {
       if (input.size() >= max_response_size_) {
-        abort("Header exceeds maximum size.");
+        abort_and_shutdown("Header exceeds maximum size.");
```

Comment:
We might have the same bug in other protocols (calling `abort` on `up` but not `shutdown`). Could you double-check this please as a followup?

Reply:
WebSocket does it correctly already. I will check LP.

Reply:
Also, `http::router` has this function:

```
void router::shutdown(const error& err) {
  abort(err);
  down_->shutdown(err);
}
```

But it's not actually implementing `shutdown`, it's basically `abort_and_shutdown`. Please rename (leave this `shutdown` in, but deprecate it so we won't break the ABI).